src/CP/valueselection/lh_utils.jl

```diff
@@ -53,46 +53,52 @@ end
 This function retrieve all the elements that are useful for doing reinforcement learning. 
 - The reward, which has been incremented through the set_reward!(PHASE, ...) functions. 
 - The terminal boolean (is the episode done or not)
-- The current state, which is the StateRepresentation of the CPModel at this stage of the solving.
+- The current state, which is the Array representation of a StateRepresentation of the CPModel at this stage of the solving.
 This state is what will then be given to the agent to make him proposed an action (a value to assign.)
+- The agent uses indexes of branchable values instead of values directly in order to facilitate the interface understanding
 - The legal_actions & legal_actions_mask used to make sure the agent won't propose a value which isn't 
 in the domain of the variable we're branching on. 
 
 The result is given as a namedtuple for convenience with ReinforcementLearning.jl interface. 
 """
 function get_observation!(lh::LearnedHeuristic, model::CPModel, x::AbstractIntVar, done = false)
+    #æet action_space_index. CAUTION : the action space is never updated. 
+    action_space_index=collect(1:size(lh.action_space)[1])
+   
     # get legal_actions_mask
     legal_actions_mask = [value in x.domain for value in lh.action_space]
 
     # compute legal actions
     legal_actions = lh.action_space[legal_actions_mask]
 
+    #get reward 
     reward = lh.reward.value
+    
     # Initialize reward for the next state: not compulsory with DefaultReward, but maybe useful in case the user forgets it
     lh.reward.value = 0
 
-    # synchronize state: we could delete env.state, we do not need it 
+    # synchronize state: Update the adjacency_matrix of the CPLayerGraph using the CPModel
```

Comment:
To be precise, the sync_state does not currently do that. This matrix is extracted when calling to_arraybuffer. The only thing that is (currently) synced here is the chosen variable and the possible values (two last columns of the array).